sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -405,14 +405,18 @@ public virtual async Task<Response<TableItem>> CreateIfNotExistsAsync(Cancellati
         /// Deletes the table specified by the tableName parameter used to construct this client instance.
         /// </summary>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
-        /// <returns></returns>
+        /// <returns>If the table exists, a <see cref="Response"/>. If the table already does not exist, <c>null</c>.</returns>
         public virtual Response Delete(CancellationToken cancellationToken = default)
         {
             using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TableClient)}.{nameof(Delete)}");
             scope.Start();
             try
             {
-                return _tableOperations.Delete(table: Name, cancellationToken: cancellationToken);
+                return _tableOperations.Delete(Name, cancellationToken);
+            }
+            catch (RequestFailedException ex) when (ex.Status == (int)HttpStatusCode.NotFound)
```

Comment:
This is still going to log an error for App Insights, I believe.   Should we consider talking to Pavel and Ted about the approach they used to try and mitigate that for Storage?

Reply:
I know @pakrym was working on a solution to this - not sure if it is ready yet.

Reply:
I would recommend moving the request execution code into this method and handling status codes yoursel.f


```
using var message = _tableOperations.CreateDeleteRequest(table);
_pipeline.Send(message, cancellationToken);

switch (message.Response.Status)
            {
                case 404:
                case 204:
                    return message.Response;
                default:
                    throw _clientDiagnostics.CreateRequestFailedException(message.Response);
            }
``